api/v1/swagger.yaml

```diff
@@ -7402,6 +7402,18 @@ paths:
           in: query
           required: false
           type: string
+        -
+          name: origin
+          description: Filter rooms by origin
+          in: query
+          required: false
+          type: string
+          enum:
+            - auto_create
+            - GUS
```

Comment:
is there any way to substitute a more generic (NON-SALESFORCE-Y) label for this?

Reply:
since you can create a room from a refocus sample, shouldn't that be an option too?

Reply:
If we're looking for more generic options then that might be an argument for free text rather than an enum, what do you think? Adding a refocus entry to the enum now

Reply:
We need to be careful here because the code is already in GUS to send the param origin=GUS. If we remove GUS out of the enum, then Refocus Rooms will not be able to be created from GUS anymore (it will fail on room create) unless GUS is released with a different origin=. I think there are 2 options:

1. Change enum to "TicketingSystem" and not release until GUS change is released
2. Change back to free text

Reply:
I was thinking that if in future we would like to have other specific origins like org62 etc that we dont want to have to keep coming up with other names for them

Reply:
Would free text result in useless data? We've set all of the different creation links for IMC rooms to include an origin="" param, so we can ensure they're coherent and should still be able to query by origin regardless of if it's an enum or not, right? 

Having the field as free text allows us to add new origins without a refocus code change and also stops the issue of the enum containing any "salesforce-y" terms or having broad values like "TicketingSystem" which going forward could make queries less granular.

Obviously free text leaves it more open for someone to input junk data, but in general it seems to fit the requirements of this field best. 
(As much as it pains me to recommend throwing away all of my enum code changes!)

Reply:
It pains me, too, but I agree.
So the values would be designated by each bot, right? So they're not "wide open" to end-users to modify?

Reply:
Exactly, they're hardcoded into places like GUS and refocus where users create rooms from, we don't have any free text fields where end users specify their own origins.